core/src/main/java/com/microsoft/applicationinsights/internal/profile/CdsProfileFetcherPolicy.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.applicationinsights.internal.profile;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+
+/**
+ * Class Responsible for configuration of Cds Profile Fetch.
+ */
+public final class CdsProfileFetcherPolicy {
+
+    /**
+     * Maximum number of instant retries to CDS to resolve ikey to AppId.
+     */
+    public int maxInstantRetries;
+
+    /**
+     * The interval in minutes for retry counters and pending tasks to be cleaned.
+     */
+    public long resetPeriodInMinutes;
+
+    /**
+     * Cached instance to be reused across SDK for CDS Profile fetch calls.
+     */
+    private static CdsProfileFetcherPolicy instance;
```

Comment:
Curious, why do we want a static instance (are you looking for singleton?) versus having a per-client policy?

Reply:
Yes, I think singleton is apt here. I will use enum instead of class to make singleton. Enums are threadsafe and best way to create singleton. 
speaking about the need, I think this policy should apply to all clients universally. I don't see a need of per client policy here.